Doc/library/codecs.rst

```diff
@@ -290,58 +290,56 @@ codec will handle encoding and decoding errors.
 Error Handlers
 ^^^^^^^^^^^^^^
 
-To simplify and standardize error handling,
-codecs may implement different error handling schemes by
-accepting the *errors* string argument. The following string values are
-defined and implemented by all standard Python codecs:
+To simplify and standardize error handling, codecs may implement different
+error handling schemes by accepting the *errors* string argument:
 
-.. tabularcolumns:: |l|L|
-
-+-------------------------+-----------------------------------------------+
-| Value                   | Meaning                                       |
-+=========================+===============================================+
-| ``'strict'``            | Raise :exc:`UnicodeError` (or a subclass);    |
-|                         | this is the default. Implemented in           |
-|                         | :func:`strict_errors`.                        |
-+-------------------------+-----------------------------------------------+
-| ``'ignore'``            | Ignore the malformed data and continue        |
-|                         | without further notice. Implemented in        |
-|                         | :func:`ignore_errors`.                        |
-+-------------------------+-----------------------------------------------+
-
-The following error handlers are only applicable to
-:term:`text encodings <text encoding>`:
+      >>> 'German ß, ♬'.encode(encoding='ascii', errors='backslashreplace')
+      b'German \\xdf, \\u266c'
+      >>> 'German ß, ♬'.encode(encoding='ascii', errors='xmlcharrefreplace')
+      b'German &#223;, &#9836;'
```

Comment:
I like the idea of examples. Would you mind to add an example for all available error handlers?

It may be interesting to add an example for surrogatepass which is an uncommon case.

Reply:
Currently there are only two REPL examples, just demonstrate this sentence:
> codecs may implement different error handling schemes by accepting the *errors* string argument

If add example for all available error handlers, will the page become ugly?
![all](https://raw.githubusercontent.com/animalize/pics/master/issues/alleh.png)

`surrogatepass` is so uncommon, maybe people who need it know it naturally.
IMHO we can just describe `surrogatepass` clearly. Most readers don't need it, so don't have to be bothered by an example.

In addition, `surrogatepass` example involves encoding algorithm, I'm afraid the readers will not see the clue from it:

```python
>>> '\uD8AA'.encode(encoding='utf-8', errors='surrogatepass')
b'\xed\xa2\xaa'
```

Reply:
> I like the idea of examples. Would you mind to add an example for all available error handlers?

I'm in agreement with adding an example for anything commonly utilized, but I don't think we should necessarily add one for all of the error handlers. 

> It may be interesting to add an example for surrogatepass which is an uncommon case.

IMO, we should try to focus on having examples for the common cases. Code examples can be very helpful, but in excess they can be distracting to readers.

Reply:
I like the idea of example errors as well. I wonder if the table is making this more cluttered. 

Perhaps something like:

```
Error Name
    Definition
    Example
```
would be more helpful.

Alternatively, a blank line between REPL examples would increase readability.

Reply:
It seems you all like the examples, I will try to plan a new layout to suit this idea.

My idea about the current change:

> If an error handleris is very easy to understand, maybe no need to give an example.
IMHO `strict`/`ignore`/`replace` are the cases.
> 
> Or if an error handler is uncommon, maybe the reader doesn't need to be disturbed by an example, we can just use the text to describe it clearly.
> IMHO these are not very common in real code: `namereplace`/`surrogateescape`/`surrogatepass`.
> 
> Then only two remaining. (`backslashreplace`/`xmlcharrefreplace`)
> These two examples also teach some Unicode knowledges imperceptibly: the value of code point, `ß` is not an ASCII character.

Reply:
@animalize 
> If an error handleris is very easy to understand, maybe no need to give an example.

I think the simple ones could still benefit from an example, just to show the basics of how it works. Even if it's fairly simple, it may not be quite as easy to understand for someone reading over the codecs documentation for the first time.

> Or if an error handler is uncommon, maybe the reader doesn't need to be disturbed by an example, we can just use the text to describe it clearly.
IMHO these are not very common in real code: namereplace/surrogateescape/surrogatepass.

> Then only two remaining. (backslashreplace/xmlcharrefreplace)
These two examples also teach some Unicode knowledges imperceptibly: the value of code point, ß is not an ASCII character.

:+1:

Reply:
> It seems you all like the examples, I will try to plan a new layout to suit this idea.

Sorry, I've been too busy recently, very intense.
I will do this when I have time.